RestaurantManagementSystem/RestaurantManagementSystem/src/Core/RestaurantRequest/Entities/RegistrationRequest.cs

```diff
@@ -58,21 +61,50 @@ public class RegistrationRequest
             return new RegistrationRequest(id, name, email, address, Contact, verificationcode, Status, DateCreated, VerificationCodeExpiresAt);
         }
 
+        private List<INotification> _domainEvents= new List<INotification>();
+        public List<INotification> DomainEvents => _domainEvents;
+        public void AddDomainEvent(INotification eventItem)
+        {
+            _domainEvents = _domainEvents ?? new List<INotification>();
+            _domainEvents.Add(eventItem);
+        }
+        public void RemoveDomainEvent(INotification eventItem)
+        {
+            _domainEvents?.Remove(eventItem);
+        }
+        public void ClearDomainEvents()
+        {
+            _domainEvents?.Clear();
+        }
+
         //check if they are equal 
-        
-        public  void VertifyCode(string providedcode)
+
+        public void VertifyCode(string providedcode)
         {
             if(this.Status != RegistrationRequestStatus.PendingVerification)
             {
                 throw new EmailIsAlreadyConfirmedException();
             }
+            if(DateTime.Now >this.VerificationCodeExpiresAt)
+            {
+                throw new CodeExpiredException();
+            }
             if (providedcode != this.VerificationCode)
             {
                 throw new WrongVerificationCodeException();
             }
             this.Status = RegistrationRequestStatus.WaitingForAdminResponse;
             this.VerificationCode = null;
         }
+        
+        public void ApproveRequest()
+        {
+            if(this.Status == RegistrationRequestStatus.WaitingForAdminResponse)
```

Comment:
what if the status is not equal to `RegistrationRequestStatus.WaitingForAdminResponse`?  nothing will happen, nothing tells if it succeeded or not.
`if(this.status != RegistrationRequestStatus.WaitingForAdminResponse) {
   // throw error
}
this.Status = RegistrationRequestStatus.Approved;
 AddDomainEvent(new RequestApprovedDomainEvent(this));
`